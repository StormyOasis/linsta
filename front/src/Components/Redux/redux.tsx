import { configureStore } from '@reduxjs/toolkit';
import { useDispatch, useSelector } from 'react-redux';

import authSliceCreator from './slices/auth.slice';
import modalSliceCreator from './slices/modals.slice';
import profileSliceCreator from './slices/profile.slice';

export type Actions = {
    authActions?: any,
    modalActions?: any;
    profileActions?: any;
}

export const actions: Actions = {};

export const buildStore = (initialState?: any) => {
    const preloadedState: any = initialState == null ? {} : initialState;
    const { authActions, authReducer } = authSliceCreator(preloadedState.auth);
    const { modalActions, modalReducer } = modalSliceCreator(preloadedState.modal);
    const { profileActions, profileReducer } = profileSliceCreator(preloadedState.profile);

    actions.authActions = authActions;
    actions.modalActions = modalActions;
    actions.profileActions = profileActions;

    const store = configureStore({
        devTools: true,
        reducer: {
            auth: authReducer,
            modal: modalReducer,
            profile: profileReducer
        },
        ...preloadedState
    });

    return store;
}

export type ConfiguredStore = ReturnType<typeof buildStore>;
export type StoreGetState = ConfiguredStore["getState"];
export type RootState = ReturnType<StoreGetState>;
export type AppDispatch = ConfiguredStore["dispatch"];

// Use throughout instead of `useDispatch` and `useSelector`
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();