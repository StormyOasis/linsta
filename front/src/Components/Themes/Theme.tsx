import React from "react";
import { ThemeProvider } from "styled-components";

const defaultTheme = {
    colors: {
        defaultLinkColor: "black",
        defaultTextColor: "black",
        backgroundColor: "white",
        backgroundDisabled: "rgba(0, 0, 0, .6)",
        borderDefaultColor: "rgb(220, 220, 220)",
        borderDarkColor: "rgb(180,180, 180)",
        inputBackgroundColor: "rgb(250, 250, 250)",
        navLinkHoverColor: "rgb(236, 236, 236)",
        buttonTextColorDefault: "white",
        buttonSecondaryTextColorDefault: "black",
        buttonDefaultColor: "rgb(0, 150, 245)",
        buttonDefaultSecondaryColor: "rgb(240, 240, 240)",
        buttonDefaultColorTrans: "rgba(0, 150, 245, .5)",
        buttonSecondaryDefaultColorTrans: "rgba(240, 240, 240, .5)",
        buttonOnHoverColor: "rgb(25, 120, 240)",        
        buttonSecondaryOnHoverColor: "rgb(220, 220, 220)",        
        inputTextColor: "rgb(115,115,115)",
        navLinkTextColor: "rgb(0, 50, 100)",
        cropperAspectBkgnd: "rgba(25,25,25,.6)",
        cropperAspectBkgndNoTrans: "rgb(25,25,25)",
        mediumTextColor: "rgb(120, 120, 120)",  
        mediaSliderButtonColor: "rgba(0,0,0, .6)",
        mediaSliderButtonBkgndColor: "rgba(225,225,225, .4)",
        warningTextColor: "rgb(235, 70, 90)"
    },    

    input: {
        backgroundColor: "#EFEFEF",        
    },

    sizes: {
        sideBarNavWidthDefault: "244px",
        sideBarNavWidthNarrow: "72px",
        sideBarNavBottomHeight: "50px",
        cropperHeight: "275px",
        defaultModalWidth: "600px",
        maxModalWidth: "75vw",
        maxModalHeight: "75vh",
        minModalHeight: "300px",
        minPostTextEditorHeight: "140px",
        maxPostTextEditorHeight: "140px",
        feedPostMinWidth: "400px",
        maxCommentModalContentHeight: "75vh",
        minCommentModalContentHeight: "300px",
    },

    breakpoints: {
        sm: "576",
        md: "768",
        lg: "1280"
    }
};

export const LexicalEditorTheme = {
    ltr: 'ltr',
    rtl: 'rtl',
};

const Theme = ({children}: any) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
};

export default Theme;
