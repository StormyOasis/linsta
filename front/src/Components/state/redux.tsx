import { configureStore } from "@reduxjs/toolkit";
import rootReducer from "./reducers/rootReducer";
import { initialState } from "./store";

const store = configureStore({
    reducer: rootReducer,
    devTools: true,
    preloadedState: initialState,
  });
  
  export type AppDispatch = typeof store.dispatch;
  export type RootState = ReturnType<typeof store.getState>;
  
  export default store;
  