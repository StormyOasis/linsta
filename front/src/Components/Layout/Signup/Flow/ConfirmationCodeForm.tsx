import React, { useEffect } from "react";
import { styled } from "styled-components";

import { getAccountsSendVerifyNotification, postAccountsAttempt } from "../../../../api/ServiceController";
import StyledInput from "../../../../Components/Common/StyledInput";
import StyledButton from "../../../../Components/Common/StyledButton";
import { useDispatch } from "react-redux";
import { AppDispatch, actions } from "../../../../Components/Redux/redux";

const ConfirmationWrapper = styled.div`
    align-content: stretch;
    align-items: stretch;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    overflow: visible;
    position: relative;
    max-width: 350px;
    padding: 8px 28px;
`;

const EmailImage = styled.span`
    background-image: url('/public/images/email.svg');
    background-repeat: none;
    display: block;
    height: 84px;
    width: 117px;
`;

const ConfirmationInnerWrapper = styled.div`
    align-content: stretch;
    align-items: stretch;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    flex-grow: 0;
    flex-shrink: 0;
    margin-bottom: 8px;
    margin-top: 16px;
    position: relative;
    overflow: visible; 
`;

const Text = styled.span`
    display: block;    
    margin: 0;
    overflow: visible;
    overflow-wrap: break-word;
    position: relative;
    text-align: center;
    text-wrap: wrap;
    word-break: break-word;
`;

const ResendButton = styled.button`
    background-color: ${props => props.theme['colors'].backgroundColor};
    color: ${props => props.theme['colors'].buttonDefaultColor};
    cursor: pointer;
    align-items: flex-start;
    position: relative;
    text-align: center;
    font-weight: 600;
    border: none;
    overflow-wrap: break-word;
    overflow: visible;
    text-wrap: wrap;
`;

const PrevButton = styled.button`
    background-color: ${props => props.theme['colors'].backgroundColor};
    color: ${props => props.theme['colors'].buttonDefaultColor};
    cursor: pointer;
    align-items: flex-start
    position: relative;
    text-align: center;
    font-weight: 600;
    border: none;
    font-size: 14px;
    display: block;
    line-height: 18px;
    overflow-wrap: break-word;
    overflow: visible;
    text-wrap: wrap;
`;

export type ConfirmationCodeFormProps = {
    userName: string;
    fullName: string;
    emailOrPhone: string;
    password: string;
    confirmationCode: string;
    month: number;
    day: number;
    year: number;
    confirmationCode_valid: boolean,
    handleFormChange: any,
    changePage: any,
}

const ConfirmationCodeForm: React.FC<ConfirmationCodeFormProps> = (props: ConfirmationCodeFormProps) => {
    const dispatch = useDispatch<AppDispatch>();
    
    useEffect(() => {
           resendCode(); //initial send
    }, [])

    function resendCode() {
        getAccountsSendVerifyNotification(props.emailOrPhone)
            .then((res: any) => res)
            .catch((err: any) => {
                console.error(err);
            });
    }

    function handleFormChange (event: React.ChangeEvent<HTMLInputElement>) {
        props.handleFormChange(
            event.target.name,
            event.target.value,
            event.target.value.length > 0
        );
    };    

    async function submitForm() {
        const result = await postAccountsAttempt({
            dryRun: false,
            userName: props.userName,
            fullName: props.fullName,
            emailOrPhone: props.emailOrPhone,
            password: props.password,
            confirmCode: props.confirmationCode,
            month: props.month,
            day: props.day,
            year: props.year,
        });

        if(result.status === 200) {                             
            const userName = props.userName;
            const password = props.password;

            dispatch(actions.authActions.login({userName, password}))
        }
    }

    return (
        <>
            <ConfirmationWrapper style={{ alignItems: "center" }}>
                <EmailImage aria-label="Email Confirmation" />
                <ConfirmationInnerWrapper>
                    <Text style={{ fontWeight: 600, margin: 0 }}>
                        Enter Confirmation Code
                    </Text>
                </ConfirmationInnerWrapper>
                <ConfirmationInnerWrapper>
                    <Text style={{ maxWidth: "100%" }}>
                        Enter the confirmation code we sent to {props.emailOrPhone}.
                        <ResendButton onClick={resendCode}>Resend Code</ResendButton>
                    </Text>
                </ConfirmationInnerWrapper>
            </ConfirmationWrapper>
            <ConfirmationWrapper>
                <ConfirmationInnerWrapper style={{ alignItems: "center", margin: "8px 0", padding: "0 40px" }}>
                    <ConfirmationWrapper style={{ margin: 0, padding: 0 }}>
                        <StyledInput style={{
                            borderRadius: "6px", cursor: "text", lineHeight: "30px",
                            padding: "0px 12px", textAlign: "left", textTransform: "none", width: "240px"}}
                            maxLength={8}
                            type="text"
                            name="confirmationCode"
                            value={props.confirmationCode}
                            placeholder="Confirmation Code"
                            onChange={handleFormChange} 
                            isValid={props.confirmationCode_valid} 
                        />
                    </ConfirmationWrapper>
                    <ConfirmationWrapper style={{ width: "100%", margin: 0, padding: "16px 8px 16px 8px" }}>
                        <StyledButton
                            datatestid="submit-signupconfirm"
                            type="button" text="Next" disabled={!props.confirmationCode_valid}
                            style={{ margin: 0 }} onClick={() => { submitForm(); }}>
                        </StyledButton>
                    </ConfirmationWrapper>
                    <ConfirmationInnerWrapper style={{ margin: 0 }}>
                        <PrevButton onClick={() => props.changePage(-1)}>Go Back</PrevButton>
                    </ConfirmationInnerWrapper>
                </ConfirmationInnerWrapper>
            </ConfirmationWrapper>
        </>
    );
}

export default ConfirmationCodeForm;