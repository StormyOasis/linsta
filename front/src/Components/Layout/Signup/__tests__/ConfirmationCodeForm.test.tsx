import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ConfirmationCodeForm, { ConfirmationCodeFormProps } from '../Flow/ConfirmationCodeForm';
import { MemoryRouter } from 'react-router-dom';
import { buildStore } from '../../../../Components/Redux/redux';
import { Provider } from 'react-redux';
import Theme from '../../../../Components/Themes/Theme';


describe("Signup Confirmation Form Component", () => {

    let props: ConfirmationCodeFormProps = {
        userName: '',
        fullName: '',
        emailOrPhone: '',
        password: '',
        confirmationCode: '',
        month: 0,
        day: 0,
        year: 0,
        confirmationCode_valid: false,
        handleFormChange: jest.fn(),
        changePage: jest.fn()
    }

    it("should render correctly", () => {
        const store = buildStore();
        const r = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Theme>
                        <ConfirmationCodeForm {...props} />
                    </Theme>
                </Provider>
            </MemoryRouter>).asFragment();
        expect(r).toMatchSnapshot();
        expect(screen.queryByTestId("submit-signupconfirm")?.hasAttribute("disabled")).toBeTruthy();
    });

    it("should validate all params and render correctly", () => {
        const oldProps = props;

        props.emailOrPhone = "a@a.com";
        props.fullName = "full name";
        props.password = "Pa$$w0rd!";
        props.userName = "username";
        props.confirmationCode_valid = true;
        props.month = 1;
        props.day = 1;
        props.year = 2011;
        props.confirmationCode = "abcde12";

        const store = buildStore();
        const r = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Theme>
                        <ConfirmationCodeForm {...props} />
                    </Theme>
                </Provider>
            </MemoryRouter>).asFragment();
        expect(r).toMatchSnapshot();
        expect(screen.queryByTestId("submit-signupconfirm")?.hasAttribute("disabled")).toBeFalsy();

        props = oldProps;
    })
});