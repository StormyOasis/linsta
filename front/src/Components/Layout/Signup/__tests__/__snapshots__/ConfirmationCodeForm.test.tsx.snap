// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Confirmation Form Component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="ConfirmationCodeForm__ConfirmationWrapper-sc-1ois2aw-0 efWDGZ"
    style="align-items: center;"
  >
    <span
      aria-label="Email Confirmation"
      class="ConfirmationCodeForm__EmailImage-sc-1ois2aw-1 cyVRga"
    />
    <div
      class="ConfirmationCodeForm__ConfirmationInnerWrapper-sc-1ois2aw-2 jZUObA"
    >
      <span
        class="ConfirmationCodeForm__Text-sc-1ois2aw-3 eactoZ"
        style="font-weight: 600; margin: 0px;"
      >
        Enter Confirmation Code
      </span>
    </div>
    <div
      class="ConfirmationCodeForm__ConfirmationInnerWrapper-sc-1ois2aw-2 jZUObA"
    >
      <span
        class="ConfirmationCodeForm__Text-sc-1ois2aw-3 eactoZ"
        style="max-width: 100%;"
      >
        Enter the confirmation code we sent to .
        <button
          class="ConfirmationCodeForm__ResendButton-sc-1ois2aw-4 dvbEtO"
        >
          Resend Code
        </button>
      </span>
    </div>
  </div>
  <div
    class="ConfirmationCodeForm__ConfirmationWrapper-sc-1ois2aw-0 efWDGZ"
  >
    <div
      class="ConfirmationCodeForm__ConfirmationInnerWrapper-sc-1ois2aw-2 jZUObA"
      style="align-items: center; margin: 8px 0px; padding: 0px 40px;"
    >
      <div
        class="ConfirmationCodeForm__ConfirmationWrapper-sc-1ois2aw-0 efWDGZ"
        style="margin: 0px; padding: 0px;"
      >
        <div
          class="StyledInput__StyledInputWrapper-sc-fkf055-0 hOmQjA"
        >
          <label
            class="StyledInput__StyledInputLabel-sc-fkf055-2 gizmBW"
          >
            <input
              class="StyledInput__StyledInputInput-sc-fkf055-1 ejdHNL"
              maxlength="8"
              name="confirmationCode"
              placeholder="Confirmation Code"
              style="border-radius: 6px; cursor: text; line-height: 30px; padding: 0px 12px; text-align: left; text-transform: none; width: 240px;"
              type="text"
              value=""
            />
          </label>
        </div>
      </div>
      <div
        class="ConfirmationCodeForm__ConfirmationWrapper-sc-1ois2aw-0 efWDGZ"
        style="width: 100%; margin: 0px; padding: 16px 8px 16px 8px;"
      >
        <button
          class="StyledButton__StyledButtonComponent-sc-8t4kac-0 fLkIsR"
          data-testid="submit-signupconfirm"
          disabled=""
          style="margin: 0px;"
          type="button"
        >
          Next
        </button>
      </div>
      <div
        class="ConfirmationCodeForm__ConfirmationInnerWrapper-sc-1ois2aw-2 jZUObA"
        style="margin: 0px;"
      >
        <button
          class="ConfirmationCodeForm__PrevButton-sc-1ois2aw-5 hihaLz"
        >
          Go Back
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Signup Confirmation Form Component should validate all params and render correctly 1`] = `
<DocumentFragment>
  <div
    class="ConfirmationCodeForm__ConfirmationWrapper-sc-1ois2aw-0 efWDGZ"
    style="align-items: center;"
  >
    <span
      aria-label="Email Confirmation"
      class="ConfirmationCodeForm__EmailImage-sc-1ois2aw-1 cyVRga"
    />
    <div
      class="ConfirmationCodeForm__ConfirmationInnerWrapper-sc-1ois2aw-2 jZUObA"
    >
      <span
        class="ConfirmationCodeForm__Text-sc-1ois2aw-3 eactoZ"
        style="font-weight: 600; margin: 0px;"
      >
        Enter Confirmation Code
      </span>
    </div>
    <div
      class="ConfirmationCodeForm__ConfirmationInnerWrapper-sc-1ois2aw-2 jZUObA"
    >
      <span
        class="ConfirmationCodeForm__Text-sc-1ois2aw-3 eactoZ"
        style="max-width: 100%;"
      >
        Enter the confirmation code we sent to a@a.com.
        <button
          class="ConfirmationCodeForm__ResendButton-sc-1ois2aw-4 dvbEtO"
        >
          Resend Code
        </button>
      </span>
    </div>
  </div>
  <div
    class="ConfirmationCodeForm__ConfirmationWrapper-sc-1ois2aw-0 efWDGZ"
  >
    <div
      class="ConfirmationCodeForm__ConfirmationInnerWrapper-sc-1ois2aw-2 jZUObA"
      style="align-items: center; margin: 8px 0px; padding: 0px 40px;"
    >
      <div
        class="ConfirmationCodeForm__ConfirmationWrapper-sc-1ois2aw-0 efWDGZ"
        style="margin: 0px; padding: 0px;"
      >
        <div
          class="StyledInput__StyledInputWrapper-sc-fkf055-0 hOmQjA"
        >
          <label
            class="StyledInput__StyledInputLabel-sc-fkf055-2 gizmBW"
          >
            <input
              class="StyledInput__StyledInputInput-sc-fkf055-1 ejdHNL"
              maxlength="8"
              name="confirmationCode"
              placeholder="Confirmation Code"
              style="border-radius: 6px; cursor: text; line-height: 30px; padding: 0px 12px; text-align: left; text-transform: none; width: 240px;"
              type="text"
              value="abcde12"
            />
          </label>
        </div>
      </div>
      <div
        class="ConfirmationCodeForm__ConfirmationWrapper-sc-1ois2aw-0 efWDGZ"
        style="width: 100%; margin: 0px; padding: 16px 8px 16px 8px;"
      >
        <button
          class="StyledButton__StyledButtonComponent-sc-8t4kac-0 jZJObc"
          data-testid="submit-signupconfirm"
          style="margin: 0px;"
          type="button"
        >
          Next
        </button>
      </div>
      <div
        class="ConfirmationCodeForm__ConfirmationInnerWrapper-sc-1ois2aw-2 jZUObA"
        style="margin: 0px;"
      >
        <button
          class="ConfirmationCodeForm__PrevButton-sc-1ois2aw-5 hihaLz"
        >
          Go Back
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
